ingest/downloader/flattener.py

```diff
@@ -0,0 +1,109 @@
+from typing import List
+
+ONTOLOGY_REQUIRED_PROPS = ['ontology', 'ontology_label']
+EXCLUDE_KEYS = ['describedBy', 'schema_type']
+
+
+class Flattener:
+    def __init__(self):
+        self.workbook = {}
+
+    def flatten(self, entity_list: List[dict], object_key: str = ''):
+        for entity in entity_list:
+            self._flatten_entity(entity, object_key)
+        return self.workbook
+
+    def _flatten_entity(self, entity, object_key):
+        worksheet_name = object_key
+        row = {}
+        content = entity
+
+        if not object_key:
+            content = entity['content']
+            worksheet_name = self._get_concrete_entity(content)
+            row = {f'{worksheet_name}.uuid': entity['uuid']['uuid']}
+
+        if not worksheet_name:
+            raise Exception('There should be a worksheet name')
+
+        self._flatten_object(content, row, parent_key=worksheet_name)
+
+        user_friendly_worksheet_name = self._format_worksheet_name(worksheet_name)
+        worksheet = self.workbook.get(user_friendly_worksheet_name, {'headers': [], 'values': []})
+
+        rows = self._update_rows(row, worksheet)
+        headers = self._update_headers(row, worksheet)
+
+        self.workbook[user_friendly_worksheet_name] = {
+            'headers': headers,
+            'values': rows
+        }
+
+    def _update_rows(self, row, worksheet):
```

Comment:
I would name this one `append_row` or even `append_row_to_sheet`

Reply:
Done.